src/MobxPromise.ts

```diff
@@ -163,9 +191,26 @@ export class MobxPromiseImpl<R>
 	@computed private get latestInvokeId()
 	{
 		window.clearTimeout(this._latestInvokeId);
-		let promise = this.invoke();
-		let invokeId:number = window.setTimeout(() => this.setPending(invokeId, promise));
-		return this._latestInvokeId = invokeId;
+		this.lastResultWasSync = this.internalSyncResultSet;
+		this.internalSyncResultSet = false;
+
+		let promise = this.invoke((result:R)=>{
+			this.internalSyncResult = result;
+			this.internalSyncResultSet = true;
+		});
+
+		if (this.internalSyncResultSet) {
+			// synchronous mode means we don't have to update anything
+			this.invokeId += 1; // have to change the value in order to trigger recomputation of users of `latestInvokeId`
+
+            if (this.onResult)
+                this.onResult(this.internalSyncResult);
```

Comment:
this will be first time onResult is called syncronously.  I think we set other observables in onResult implementations.  Just flagging a potential issue we should check

Reply:
I guess to get around that we could just fire onResult in a setTimeout, nbd